chainer/functions/math/sparse_matmul.py

```diff
@@ -0,0 +1,512 @@
+import numpy
+
+import chainer
+from chainer.backends import cuda
+from chainer import function_node
+from chainer import utils
+from chainer.utils import type_check
+
+try:
+    from scipy import sparse
+    _scipy_available = True
+except ImportError:
+    _scipy_available = False
+
+
+class SparseCooMatrix(object):
+
+    def __init__(self, data, row, col, shape, use_variable=False):
```

Comment:
When is `use_variable=True` needed? In my understanding, gradients are not propagated to the sparse matrix.

Reply:
It is used when grads of sparse matrix are necessary as you thought.  I've heard from a few researchers that there are applications in which grads of sparse matrix are needed.

Reply:
To what extent do you think we should support it (in this PR)? I talked with @beam2d and concluded that we could wait with this feature, because we need to think though the interface and maybe also include differentiable conversions (`FunctionNode` implementations) between dense and sparse representations?

Reply:
Dose that mean that you would not rush to conclude on how to handle gradients of sparse matrix? I understand it. So, should I delete codes that are related to this feature like `SparseMatMulGrandSP` among others from this PR?

Reply:
No exactly, that's what we thought. But I am not entirely sure how gradients of `A` in `A(sparse) * B (dense) = C` should be defined. In theory, they become dense and thus `A` after applying an update is no longer a sparse matrix, but this probably depends on the use case. Do you have other sources?

Reply:
For the record, MXNet supports [sparse gradient](http://pytorch.org/docs/master/sparse.html) it seems. PyTorch have a sparse module that's experimental and not that well documented. I'm not sure about TensorFlow but there are discussions with varying conclusions. TVM mentions nothing about sparse representations.